File: Data Structures & Algorithms/number-of-one-bits/submission-0.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        position = 31
        while position >= 0:
            if n >= 2**position:
                print(position)
                n -= 2**position
                count +=1
            position -= 1
        
        return count

File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> wordsPerAnagram = new HashMap<>();

        for (String elem: strs){
            int [] count = new int[26];
            for (char c: elem.toCharArray()){
                count[c - 'a'] ++;
            }
            
            String key = Arrays.toString(count);
            if (!wordsPerAnagram.containsKey(key)){
                wordsPerAnagram.put(key, new ArrayList<>());
            } 
            wordsPerAnagram.get(key).add(elem);
        }

        //System.out.println((wordsPerAnagram.values()));
        return new ArrayList<>(wordsPerAnagram.values());
    }
}
